Deprecation errors when moving to Ruby 3.0

When upgrading to Ruby 3.0 I get the following errors when trying to  build my site:

```
11:20:54 AM: ────────────────────────────────────────────────────────────────
11:20:54 AM:   1. Build command from Netlify app                             
11:20:54 AM: ────────────────────────────────────────────────────────────────
11:20:54 AM: ​
11:20:54 AM: $ jekyll build; npm run workbox generateSW workbox-config.js;
11:20:55 AM: Configuration file: /opt/build/repo/_config.yml
11:20:55 AM:             Source: /opt/build/repo
11:20:55 AM:        Destination: /opt/build/repo/_site
11:20:55 AM:  Incremental build: disabled. Enable with --incremental
11:20:55 AM:       Generating...
11:20:55 AM:        Jekyll Feed: Generating feed for posts
11:20:55 AM:                     ------------------------------------------------
11:20:55 AM:       Jekyll 4.0.1   Please append `--trace` to the `build` command 
11:20:55 AM:                      for any additional information or backtrace. 
11:20:55 AM:                     ------------------------------------------------
11:20:55 AM: /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:190:in `_fill_front_matter_menu': wrong number of arguments (given 2, expected 1) (ArgumentError)
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:130:in `block in _front_matter_menu'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:110:in `each'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:110:in `_front_matter_menu'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:69:in `block (2 levels) in _page_menus'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:68:in `map'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:68:in `block in _page_menus'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:67:in `each'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:67:in `each_with_object'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:67:in `_page_menus'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:18:in `menus'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus.rb:26:in `to_liquid_drop'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-menus-0.6.1/lib/jekyll/menus/hook.rb:14:in `block in <top (required)>'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/hooks.rb:102:in `block in trigger'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/hooks.rb:101:in `each'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/hooks.rb:101:in `trigger'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/site.rb:199:in `render'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/site.rb:76:in `process'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/command.rb:28:in `process_site'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/commands/build.rb:65:in `build'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/commands/build.rb:36:in `process'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/command.rb:89:in `block in process_with_graceful_fail'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/command.rb:89:in `each'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/command.rb:89:in `process_with_graceful_fail'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/lib/jekyll/commands/build.rb:18:in `block (2 levels) in init_with_program'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
	from /opt/build/cache/bundle/ruby/3.0.0/gems/jekyll-4.0.1/exe/jekyll:15:in `<top (required)>'
	from /opt/build/cache/binstubs/jekyll:29:in `load'
	from /opt/build/cache/binstubs/jekyll:29:in `<main>'
11:20:55 AM: > @ workbox /opt/build/repo
11:20:55 AM: > workbox "generateSW" "workbox-config.js"
11:20:56 AM: Using configuration from /opt/build/repo/workbox-config.js.
11:20:56 AM: Service worker generation failed:
11:20:56 AM: 
11:20:56 AM: Couldn't find configuration for either precaching or runtime caching. Please ensure that the various glob options are set to match one or more files, and/or configure the runtimeCaching option.
    at module.exports (/opt/build/repo/node_modules/workbox-build/build/lib/populate-sw-template.js:41:11)
11:20:56 AM:     at module.exports (/opt/build/repo/node_modules/workbox-build/build/lib/write-sw-using-default-template.js:50:25)
11:20:56 AM:     at async Object.generateSW (/opt/build/repo/node_modules/workbox-build/build/generate-sw.js:229:21)
11:20:56 AM:     at async runBuildCommand (/opt/build/repo/node_modules/workbox-cli/build/app.js:29:54)
11:20:56 AM:     at async Object.exports.app (/opt/build/repo/node_modules/workbox-cli/build/app.js:105:17)
11:20:56 AM: 
11:20:56 AM: npm ERR! code ELIFECYCLE
11:20:56 AM: npm ERR! errno 1
11:20:56 AM: npm ERR! @ workbox: `workbox "generateSW" "workbox-config.js"`
11:20:56 AM: npm ERR! Exit status 1
11:20:56 AM: npm ERR!
11:20:56 AM: npm ERR! Failed at the @ workbox script.
11:20:56 AM: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
11:20:56 AM: npm ERR! A complete log of this run can be found in:
11:20:56 AM: npm ERR!     /opt/buildhome/.npm/_logs/2021-06-29T17_20_56_767Z-debug.log
11:20:56 AM: ​
11:20:56 AM: ────────────────────────────────────────────────────────────────
11:20:56 AM:   "build.command" failed                                        
11:20:56 AM: ────────────────────────────────────────────────────────────────
11:20:56 AM: ​
11:20:56 AM:   Error message
11:20:56 AM:   Command failed with exit code 1: jekyll build; npm run workbox generateSW workbox-config.js;
11:20:56 AM: ​
11:20:56 AM:   Error location
11:20:56 AM:   In Build command from Netlify app:
11:20:56 AM:   jekyll build; npm run workbox generateSW workbox-config.js;
11:20:56 AM: ​
11:20:56 AM:   Resolved config
11:20:56 AM:   build:
11:20:56 AM:     command: jekyll build; npm run workbox generateSW workbox-config.js;
11:20:56 AM:     commandOrigin: ui
11:20:56 AM:     environment:
11:20:56 AM:       - JEKYLL_ENV
11:20:56 AM:       - RUBY_VERSION
11:20:56 AM:     publish: /opt/build/repo/_site
11:20:56 AM:     publishOrigin: ui
11:20:56 AM:   functions:
11:20:56 AM:     '*': {}
11:20:57 AM: Caching artifacts
11:20:57 AM: Started saving ruby gems
11:20:57 AM: Finished saving ruby gems
11:20:57 AM: Started saving node modules
11:20:57 AM: Finished saving node modules
11:20:57 AM: Started saving build plugins
```

--- a/lib/jekyll/menus.rb
+++ b/lib/jekyll/menus.rb
@@ -116,7 +116,7 @@ module Jekyll
           # --
 
           if item.is_a?(String)
-            out[key] << _fill_front_matter_menu({ "identifier" => item }, {
+            out[key] << _fill_front_matter_menu({ "identifier" => item }, **{
               :page => page
             })
 
@@ -127,7 +127,7 @@ module Jekyll
           # --
 
           elsif item.is_a?(Hash)
-            out[key] << _fill_front_matter_menu(item, {
+            out[key] << _fill_front_matter_menu(item, **{
               :page => page
             })
 
@@ -178,7 +178,7 @@ module Jekyll
 
       else
         mergeable[menu] ||= []
-        mergeable[menu] << _fill_front_matter_menu(nil, {
+        mergeable[menu] << _fill_front_matter_menu(nil, **{
           :page => page
         })
       end

